File: data/src/main/java/com/mrp/sml/data/local/TransferDao.java
package com.mrp.sml.data.local;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import java.util.List;

@Dao
public interface TransferDao {
    @Query("SELECT * FROM transfer_history ORDER BY timestampEpochMillis DESC")
    List<TransferEntity> getTransferHistory();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(TransferEntity record);
}

File: data/src/main/kotlin/com/mrp/sml/data/local/TransferDao.kt
package com.mrp.sml.data.local

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import kotlinx.coroutines.flow.Flow

@Dao
interface TransferDao {
    @Query("SELECT * FROM transfer_history ORDER BY timestampEpochMillis DESC")
    fun observeTransferHistory(): Flow<List<TransferEntity>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insert(record: TransferEntity)
}
